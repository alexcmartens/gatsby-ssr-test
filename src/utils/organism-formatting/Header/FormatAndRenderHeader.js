import React from "react"

import renderPaddingOrFullBleed from "../renderPaddingOrFullBleed"
import { Header } from "terra-one"

const FormatAndRenderHeader = ({ data }) => {
  return (
    <>
      {renderPaddingOrFullBleed(
        <Header
          eyebrow={data?.eyebrow}
          headline={data?.headline}
          subhead={data?.subhead}
        />
      )}
    </>
  )
}

export default FormatAndRenderHeader
