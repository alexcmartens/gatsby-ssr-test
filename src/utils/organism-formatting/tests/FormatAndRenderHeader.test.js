import React from "react"
import renderer from "react-test-renderer"
import FormatAndRenderHeader from "../Header/FormatAndRenderHeader"

describe("FormatAndRenderHeader", () => {
  it("renders correctly", () => {
    const data = {
      eyebrow: "Eyebrow option",
      headline: "Header title",
      subhead: "Subhead can go here.",
    }

    const tree = renderer.create(<FormatAndRenderHeader data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
