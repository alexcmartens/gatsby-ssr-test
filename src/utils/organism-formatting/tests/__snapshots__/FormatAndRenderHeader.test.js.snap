// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatAndRenderHeader renders correctly 1`] = `
<section
  className="default-layout   mockconstructor
      "
  style={
    Object {
      "backgroundColor": undefined,
      "marginBottom": "",
    }
  }
>
  <section
    className="default-layout__grid 
          
        "
    style={
      Object {
        "backgroundColor": "",
        "marginBottom": "",
      }
    }
  >
    <div
      className="default-layout__children "
    >
      <div
        eyebrow="Eyebrow option"
        headline="Header title"
        subhead="Subhead can go here."
      />
    </div>
  </section>
</section>
`;
